app/screens/channel/channel_title/channel_guest_label/index.tsx

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {View} from 'react-native';
+
+import FormattedText from '@components/formatted_text';
+import {General} from '@constants';
+import {t} from '@utils/i18n';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {isGuest as isTeammateGuest} from '@utils/user';
+
+type ChannelGuestLabelProps = {
+    canHaveSubtitle: boolean;
+    channelType: string;
+    guestCount: number;
+    teammateRoles: string;
+    theme: Theme;
+}
+
+const ChannelGuestLabel = ({teammateRoles, canHaveSubtitle, guestCount, channelType, theme}: ChannelGuestLabelProps) => {
+    const style = getStyle(theme);
+    const hasGuests = guestCount > 0;
+
+    const isGuest = channelType === General.DM_CHANNEL && isTeammateGuest(teammateRoles);
+
+    if (!canHaveSubtitle || (!isGuest && !hasGuests) || (channelType === General.DM_CHANNEL && !isGuest)) {
+        return null;
+    }
+
```

Comment:
Can we validate this before mounting the component in the parent component? If we cannot is fine to leave it like this

Reply:
Done